cores/esp32/esp32-hal-rmt.c

```diff
@@ -290,84 +362,37 @@ bool rmtWrite(rmt_obj_t* rmt, rmt_data_t* data, size_t size)
     }
     
     int channel = rmt->channel;
-    int allocated_size = MAX_DATA_PER_CHANNEL * rmt->buffers;
-
-    if (size > (allocated_size - 1)) {
-
-        int half_tx_nr = MAX_DATA_PER_ITTERATION/2;
-        RMT_MUTEX_LOCK(channel);
-        // setup interrupt handler if not yet installed for half and full tx
-        if (!intr_handle) {
-            esp_intr_alloc(ETS_RMT_INTR_SOURCE, (int)ARDUINO_ISR_FLAG, _rmt_isr, NULL, &intr_handle);
-        }
-
-        rmt->data_size = size - MAX_DATA_PER_ITTERATION;
-        rmt->data_ptr = ((uint32_t*)data) + MAX_DATA_PER_ITTERATION;
-        rmt->intr_mode = E_TX_INTR | E_TXTHR_INTR;
-        rmt->tx_state = E_SET_CONTI | E_FIRST_HALF;
-
-#if CONFIG_IDF_TARGET_ESP32C3
-        //uint32_t val = RMT.tx_conf[channel].val;
-        // init the tx limit for interruption
-        RMT.tx_lim[channel].limit = half_tx_nr+2;
-        //RMT.tx_conf[channel].val = val;
-        //RMT.tx_conf[channel].conf_update = 1;
-        // reset memory pointer
-        RMT.tx_conf[channel].mem_rst = 1;
-        //RMT.tx_conf[channel].mem_rst = 0;
-        RMT.tx_conf[channel].mem_rd_rst = 1;
-        //RMT.tx_conf[channel].mem_rd_rst = 0;
-#else
-        // init the tx limit for interruption
-        RMT.tx_lim_ch[channel].limit = half_tx_nr+2;
-        // reset memory pointer
-        RMT.conf_ch[channel].conf1.apb_mem_rst = 1;
-        RMT.conf_ch[channel].conf1.apb_mem_rst = 0;
-        RMT.conf_ch[channel].conf1.mem_rd_rst = 1;
-        RMT.conf_ch[channel].conf1.mem_rd_rst = 0;
-        RMT.conf_ch[channel].conf1.mem_wr_rst = 1;
-        RMT.conf_ch[channel].conf1.mem_wr_rst = 0;
-#endif
-        // set the tx end mark
-        //RMTMEM.chan[channel].data32[MAX_DATA_PER_ITTERATION].val = 0;
-
-        // clear and enable both Tx completed and half tx event
-        RMT.int_clr.val = _INT_TX_END(channel);
-        RMT.int_clr.val = _INT_THR_EVNT(channel);
-        RMT.int_clr.val = _INT_ERROR(channel);
-        
-        RMT.int_ena.val |= _INT_TX_END(channel);
-        RMT.int_ena.val |= _INT_THR_EVNT(channel);
-        RMT.int_ena.val |= _INT_ERROR(channel);
-
-        RMT_MUTEX_UNLOCK(channel);
-
-        // start the transation
-        //return _rmtSendOnce(rmt, data, MAX_DATA_PER_ITTERATION, true);
-        return _rmtSendOnce(rmt, data, MAX_DATA_PER_ITTERATION, false);
-    } else {
-        // use one-go mode if data fits one buffer 
-        return _rmtSendOnce(rmt, data, size, false);
-    }
+    RMT_MUTEX_LOCK(channel);
+    ESP_ERROR_CHECK(rmt_tx_stop(channel));
```

Comment:
`ESP_ERROR_CHECK` is also an IDF API that boils down to `ESP_LOGx`

Reply:
Actually ```ESP_ERROR_CHECK``` boils down directly to ```esp_rom_printf()```, which seems to be a ROM function, thus, it should be fine for Arduino. What do you say?

https://github.com/espressif/esp-idf/blob/master/components/esp_common/include/esp_err.h#L121-L127
https://github.com/espressif/esp-idf/blob/master/components/esp_system/esp_err.c#L23-L33

Reply:
what it looks like is that it will abort and produce some IDF formatted message. Let's stick to Arduino constructs and log_e if not ESP_OK

Reply:
OK. Done. 
I removed ```ESP_ERROR_CHECK()``` from IDF function calls that only checked IDF parameters and left it on the function call that really could return error on IDF operations.